src/model_data.rs

```diff
@@ -1,70 +1,41 @@
+use std::collections::HashMap;
+
 use crate::matrix::{size::*, Matrix, SizeMarker};
 use crate::Error;
-use serde::{de::DeserializeOwned, Deserialize, Serialize};
-use std::collections::HashMap;
-use std::io::{BufReader, Read};
 
-#[derive(Deserialize, Debug)]
+pub mod models {
+    use super::*;
+    use crate::matrix::Matrix;
+
+    include!(concat!(env!("OUT_DIR"), "/models.rs"));
+}
+
+#[derive(Debug)]
 pub struct ModelData<I, O, N> {
```

Comment:
maybe `<I, N, O>` ?

Reply:
Yeah that makes more sence since the data travels in that direction